### PR TITLE
chore: packages improvements

### PR DESCRIPTION
**Pull request by Lymdun**
- upgrades packages to their latest versions
- adds support to jetbrains rider ide
- fixes tests to support operating system with a different numerical symbol (, or .)

### CODE REVIEW

File: Assets/Tests/SocketLayer/AckSystem/AckSystemTest_Reliable.cs [hunk 1]
```diff
@@ -168,22 +168,22 @@ public void TearDown()
...
+        [TestCase(true, 3000, 0.2f, 0f, ExpectedResult = null)]
+        [TestCase(true, 3000, 0.2f, 0.4f, ExpectedResult = null)]
+        [TestCase(false, 100, 0f, 0f, ExpectedResult = null)]
+        [TestCase(false, 100, 0.2f, 0f, ExpectedResult = null)]
+        [TestCase(false, 100, 0.2f, 0.4f, ExpectedResult = null)]
+        [TestCase(false, 3000, 0.2f, 0f, ExpectedResult = null)]
+        [TestCase(false, 3000, 0.2f, 0.4f, ExpectedResult = null)]
         [Repeat(100)]
-        public void AllMessagesShouldHaveBeenReceivedInOrder(bool instance2Sends, int messageCount, float dropChance, float skipChance)
+        public IEnumerator AllMessagesShouldHaveBeenReceivedInOrder(bool instance2Sends, int messageCount, float dropChance, float skipChance)
```
Comment (James-Frowen) [inline Assets/Tests/SocketLayer/AckSystem/AckSystemTest_Reliable.cs:182]: what are these changes for?

Comment (James-Frowen) [reply Assets/Tests/SocketLayer/AckSystem/AckSystemTest_Reliable.cs:182]: maybe the `[UnityTest]` just needs to be changed to NUnit `[Test]`

File: Assets/Tests/Weaver/BitAttributeTests.cs [hunk 2]
```diff
@@ -179,22 +179,22 @@ public void Vector3PackInvalid()
...
+            HasError($"Max must be above 0, max:({-1.0:0.0}, {0.0:0.0}, {0.0:0.0})",
                "UnityEngine.Vector3 BitAttributeTests.Vector3PackInvalid.MyBehaviour::value8");
-            HasError("Max must be above 0, max:(1.0, -1.0, 0.0)",
+            HasError($"Max must be above 0, max:({1.0:0.0}, {-1.0:0.0}, {0.0:0.0})",
                "UnityEngine.Vector3 BitAttributeTests.Vector3PackInvalid.MyBehaviour::value9");
-            HasError("Max must be above 0, max:(1.0, 1.0, -1.0)",
+            HasError($"Max must be above 0, max:({1.0:0.0}, {1.0:0.0}, {-1.0:0.0})",
                "UnityEngine.Vector3 BitAttributeTests.Vector3PackInvalid.MyBehaviour::value10");
-            HasError("Max must be above 0, max:(-1.0, 0.0, 0.0)",
+            HasError($"Max must be above 0, max:({-1.0:0.0}, {0.0:0.0}, {0.0:0.0})",
```
Comment (James-Frowen) [inline Assets/Tests/Weaver/BitAttributeTests.cs:197]: Can you add a small comment about why the string interpolation, just so we dont forget

File: Assets/Tests/SocketLayer/AckSystem/AckSystemTest_Reliable.cs (hunk 1 above)
Comment (Lymdun) [reply Assets/Tests/SocketLayer/AckSystem/AckSystemTest_Reliable.cs:182]: > what are these changes for?

They are related to a change to the test framework for `[UnityTest]` tests with parameters, which requires them to returns an IEnumerator
See https://docs.unity3d.com/Packages/com.unity.test-framework@1.1/changelog/CHANGELOG.html



> maybe the `[UnityTest]` just needs to be changed to NUnit `[Test]`

Good idea, ty!

File: Assets/Tests/Weaver/BitAttributeTests.cs (hunk 2 above)
Comment (Lymdun) [reply Assets/Tests/Weaver/BitAttributeTests.cs:200]: Done

**Pull request merged**